index/secondary-index/src/test/scala/org/apache/carbondata/spark/testsuite/secondaryindex/TestSIWithSecondryIndex.scala

```diff
@@ -61,6 +62,16 @@ class TestSIWithSecondryIndex extends QueryTest with BeforeAndAfterAll {
     }
   }
 
+  test ("test alter drop columns on the SI columns") {
+    sql("create table test1 (name string, id string, country string) stored as carbondata")
+    sql("insert into test1 select 'xx', '1', 'china'")
+    sql("create index t_index on table test1(id, country) as 'carbondata'")
+    // alter table to drop the columns used in index
+    sql("alter table test1 drop columns(id, country)")
+    sql("insert into test1 select 'xy'")
+    assert(sql("show indexes on test1").collect().isEmpty)
+  }
+
   test("Test secondry index data count") {
```

Comment:
add one more test case where the all columns of are not dropped and we get the error.

Reply:
added